clustered/server/service/src/main/java/org/ehcache/clustered/server/state/EhcacheStateServiceProvider.java

```diff
@@ -73,7 +76,8 @@ public boolean initialize(ServiceProviderConfiguration configuration, PlatformCo
         LOGGER.warn("No offheap-resource defined - this will prevent provider from offering any EhcacheStateService.");
       }
     } else {
-      throw new UnsupportedOperationException("There are no offheap-resource defined, this is not supported");
+      offHeapResourcesProvider = new OffHeapResourcesProvider(emptyMap());
+      LOGGER.warn("No offheap-resource defined - this will prevent provider from offering any EhcacheStateService.");
```

Comment:
@chrisdennis : I think the warnings are misleading... There are 2 choices here:

1) either we log a warning and `return false` to tell core to discard the service
2) or we keep it as-is and fake an empty offheap, the service will stay and offer offheap services with inexisting resources...

What's the right thing for Ehcache ?

Reply:
Forget my comment ;-) since we can add offheap at runtime we cannot discard the service.